proposals/js-promise-integration/Overview.md

```diff
@@ -2,299 +2,225 @@
+1. Otherwise, returns the result to the caller:
+    * If `result` is a normal value, return as value of *suspending function*
+    * If `result` is an exception, throw exception.
+
+## Frequently Asked Questions
 
-   The `Suspender` object is used to connect a `Promise` returning import with a `Promise` returning export. Without this explicit connection, it becomes problematic especially when constructing so-called chains of modules: where one module calls into the exports of another.
+1. **Why do we prevent JavaScript programs from using this API?**
 
-   A further issue arises from potential _misuse_ of the imported function. Since imports can be immediately exported when a WebAsembly module is instantiated, a marked import can also be exported. However, without being passed an explicit capability to suspend&mdash;in the form of the connecting suspender object&mdash;any attempt to use the marked function in a different setting will not compromise the integrity of the export/import pair.
-   
-   In fact, such a use would either be completely benign&mdash;because it was used in a different module for that module's asynchronous imports&mdash;or it would simply trap or not validate&mdash;because the marked import requires a suitable `externref` in order to be callable.
+   JavaScript already has a way of managing computations that can suspend. This is semantically connected to JavaScript `Promise` objects and the `async` function syntax. However, a more important reason is that we do not wish to inadvertently introduce features that can affect the behavior of existing JavaScript programs.
 
-1. **Why do we try to prevent JavaScript programs from using this API?**
+1. **Why does `WebAssembly.promising` return a JavaScript `Function`**?
 
-   JavaScript already has a way of managing computations that can suspend. This is semantically connected to JavaScript `Promise` objects and the `async` function syntax. However, a more important reason is that it is important that we do not inadvertently introduce JavaScript language features that can affect the behavior of existing programs.
+   This allows us to be more precise in a few special cases. In particular, if two WebAssembly modules (Modules A & B) are *chained together* (e.g., the import of module A is provided by the export of module B) then we need precision about the interaction with JSPI:
+    1. If the link is composed of a `Suspending`/`promising` pair (i.e., the module A import is wrapped with `new WebAssembly.Suspending` and the module B export is wrapped with `WebAssembly.promising`) then, if module B suspends (via one of its imports), then module A will also suspend. However, there will also be an additional `Promise`: created from the wrapped export from module B.
+    1. If the link is direct -- without a `promising`/`suspending` pair -- then the connect is transparent from the perspective of JSPI; provided that there are no JavaScript function calls in the link.
```

Comment:
s/suspending/Suspending/

Reply:
ack.
Also rewriting `promising`/`suspending` with `Suspending`/`promising`, since this is referring to a link between modules.